homeassistant/components/opnsense/__init__.py

```diff
@@ -38,43 +37,50 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the OPNsense component."""
+    hass.data[DATA_HASS_CONFIG] = config
+    if config.get(DOMAIN) is not None:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the opnsense component."""
+    return True
 
-    conf = config[DOMAIN]
-    url = conf[CONF_URL]
-    api_key = conf[CONF_API_KEY]
-    api_secret = conf[CONF_API_SECRET]
-    verify_ssl = conf[CONF_VERIFY_SSL]
-    tracker_interfaces = conf[CONF_TRACKER_INTERFACE]
 
-    interfaces_client = diagnostics.InterfaceClient(
-        api_key, api_secret, url, verify_ssl, timeout=20
-    )
-    try:
-        interfaces_client.get_arp()
-    except APIException:
-        _LOGGER.exception("Failure while connecting to OPNsense API endpoint")
-        return False
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the OPNsense from a config entry."""
 
-    if tracker_interfaces:
-        # Verify that specified tracker interfaces are valid
-        netinsight_client = diagnostics.NetworkInsightClient(
-            api_key, api_secret, url, verify_ssl, timeout=20
-        )
-        interfaces = list(netinsight_client.get_interfaces().values())
-        for interface in tracker_interfaces:
-            if interface not in interfaces:
-                _LOGGER.error(
-                    "Specified OPNsense tracker interface %s is not found", interface
-                )
-                return False
+    api_data = {
+        CONF_API_KEY: entry.data[CONF_API_KEY],
+        CONF_API_SECRET: entry.data[CONF_API_SECRET],
+        "base_url": entry.data[CONF_URL],
+        "verify_cert": entry.data[CONF_VERIFY_SSL],
```

Comment:
Shouldn't `base_url` and `verify_cert` added to `homeassistant/components/opnsense/const.py`?